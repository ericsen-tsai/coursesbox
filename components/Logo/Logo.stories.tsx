import { ComponentStoryObj, ComponentMeta } from '@storybook/react'

import { expect } from '@storybook/jest'
import { screen, userEvent } from '@storybook/testing-library'

import Logo from '.'

export default {
  title: 'Content/Logo',
  component: Logo,
} as ComponentMeta<typeof Logo>

export const BasicLogo: ComponentStoryObj<typeof Logo> = {
  play: async () => {
    await expect(screen.getByRole('banner')).toBeInTheDocument()
  },
  args: {
    children: 'CoursesBox',
  },
}
