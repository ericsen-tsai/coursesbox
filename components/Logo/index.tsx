import { css } from '@emotion/react'
import styled from '@emotion/styled'

export type Props = {
  /** Logo size in rem */
  size?: number
}

const Logo = styled.header<Props>`
  font-family: 'Monoton', cursive;
  font-size: ${({ size = 3 }) => `${size}rem`};
  font-weight: 400;
  ${({ theme, size = 3 }) => {
    return css`
      color: ${theme.font.logo};
      text-shadow: 0 0 ${0.1 * size}rem ${theme.font.logoShadow1},
        0 0 ${0.05 * size}rem ${theme.font.logoShadow2},
        0 0 ${0.07 * size}rem ${theme.font.logoShadow2},
        0 0 ${0.08 * size}rem ${theme.font.logoShadow2},
        0 0 ${0.1 * size}rem ${theme.font.logoShadow2};
    `
  }};
`

export default Logo
