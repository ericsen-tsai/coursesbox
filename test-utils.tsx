import React, { FC, ReactElement, ReactNode } from 'react'

import { render, RenderOptions } from '@testing-library/react'

import { Themes } from './styles/themes'
import { ThemeProvider } from '@emotion/react'

const Wrapper = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={Themes.light}>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: Wrapper, ...options })

export * from '@testing-library/react'

export { customRender as render }
