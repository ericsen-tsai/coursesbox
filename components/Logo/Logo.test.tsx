import { render } from '@/test-utils'

import Logo from '.'

describe('Logo test cases', () => {
  it('Logo render check', () => {
    const { asFragment } = render(<Logo>CoursesBox</Logo>)

    expect(asFragment()).toMatchSnapshot()
  })

  it('Logo render with custom size', () => {
    const { asFragment } = render(<Logo size={10}>CoursesBox</Logo>)

    expect(asFragment()).toMatchSnapshot()
  })
})
