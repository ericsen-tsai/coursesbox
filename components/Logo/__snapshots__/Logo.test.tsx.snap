// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo test cases Logo render check 1`] = `
<DocumentFragment>
  <header
    class="css-i1a01a"
  >
    CoursesBox
  </header>
</DocumentFragment>
`;

exports[`Logo test cases Logo render with custom size 1`] = `
<DocumentFragment>
  <header
    class="css-1bptyt9"
    size="10"
  >
    CoursesBox
  </header>
</DocumentFragment>
`;
